includes/class-wc-stripe-blocks-support.php

```diff
@@ -0,0 +1,345 @@
+use Automattic\WooCommerce\Blocks\Payments\Api as PaymentMethodsAPI;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * WC_Stripe_Blocks_Support class.
+ *
+ * @extends AbstractPaymentMethodType
+ */
+final class WC_Stripe_Blocks_Support extends AbstractPaymentMethodType {
+	/**
+	 * Payment method name defined by payment methods extending this class.
+	 *
+	 * @var string
+	 */
+	protected $name = 'stripe';
+
+	/**
+	 * Constructor
+	 */
+	public function __construct() {
+		add_action( 'woocommerce_rest_checkout_process_payment_with_context', [ $this, 'add_payment_request_order_meta' ], 8, 2 );
+		add_action( 'woocommerce_rest_checkout_process_payment_with_context', [ $this, 'add_stripe_intents' ], 9999, 2 );
+	}
+
+	/**
+	 * Initializes the payment method type.
+	 */
+	public function initialize() {
+		$this->settings = get_option( 'woocommerce_stripe_settings', [] );
+	}
+
+	/**
+	 * Returns if this payment method should be active. If false, the scripts will not be enqueued.
+	 *
+	 * @return boolean
+	 */
+	public function is_active() {
+		return ! empty( $this->settings['enabled'] ) && 'yes' === $this->settings['enabled'];
+	}
+
+	/**
+	 * Returns an array of scripts/handles to be registered for this payment method.
+	 *
+	 * @return array
+	 */
+	public function get_payment_method_script_handles() {
+		$asset_path = WC_STRIPE_PLUGIN_PATH . '/build/index.asset.php';
+		$version = WC_STRIPE_VERSION;
+		$dependencies = [];
+		if ( file_exists( $asset_path ) ) {
+			$asset = require $asset_path;
+			$version = is_array( $asset ) && isset( $asset['version'] )
+				? $asset['version']
+				: $version;
+			$dependencies = is_array( $asset ) && isset( $asset['dependencies'] )
+				? $asset['dependencies']
+				: $dependencies;
+		}
+		wp_register_script(
+			'wc-stripe-blocks-integration',
+			WC_STRIPE_PLUGIN_URL . '/build/index.js',
+			$dependencies,
+			$version,
+			true
+		);
+
+		return [ 'wc-stripe-blocks-integration' ];
+	}
+
+	/**
+	 * Returns an array of key=>value pairs of data made available to the payment methods script.
+	 *
+	 * @return array
+	 */
+	public function get_payment_method_data() {
+		return [
+			'stripeTotalLabel'    => $this->get_total_label(),
+			'publicKey'           => $this->get_publishable_key(),
+			'allowPrepaidCard'    => $this->get_allow_prepaid_card(),
+			'title'               => $this->get_title(),
+			'button'              => [
+				'type'   => $this->get_button_type(),
+				'theme'  => $this->get_button_theme(),
+				'height' => $this->get_button_height(),
+				'locale' => $this->get_button_locale(),
+			],
+			'inline_cc_form'      => $this->get_inline_cc_form(),
+			'icons'               => $this->get_icons(),
+			'showSavedCards'      => $this->get_show_saved_cards(),
+			'allowPaymentRequest' => $this->get_allow_payment_request(),
+			'showSaveOption'      => $this->get_show_save_option(),
+			'supports'            => $this->get_supported_features(),
+		];
```

Comment:
Some of these are already available in the JS on the `wc_stripe_params` object, [see here](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/0c182f1dc17614207c5c559abcae220b9735882e/includes/class-wc-gateway-stripe.php#L418-L470).

List of already available things in the format `[this PR] -> [pre-existing]`

- `publicKey -> key`
- `allowPrepaidCard -> allow_prepaid_card`
- `inline_cc_form -> inline_cc_form`

I believe all the payment request button options are already available on `wc_stripe_payment_request`, [see here](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/0c182f1dc17614207c5c559abcae220b9735882e/includes/payment-methods/class-wc-stripe-payment-request.php#L507-L548).

I wonder if it'd be better to rely on the pre-existing values — i.e. implicitly trusting that we're not going to change them down the line — or add the new ones here.

I'm personally leaning towards keeping this as is, just to avoid any future problems where we might change the name of a key, remove keys we think have become redundant, etc. I could be convinced otherwise though.

There's one definite bad thing about registering these variables twice; if we change any available filters, remembering to do that in both (eventually more than 2 maybe?) places will be difficult.

Maybe we should have the [main `scripts()` function](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/0c182f1dc17614207c5c559abcae220b9735882e/includes/payment-methods/class-wc-stripe-payment-request.php#L484) detect whether the blocks are loaded, and load in any extras needed for the blocks?

I'd love to hear what you think @nerrad. 
Same goes for others in Fractal, but I'll ping them in Slack so they don't get pinged for every discussion happening in the PR.

Reply:
This is a place where it'd make sense to try and abstract things shared between implementations of Stripe so that there's no duplication (similarly to what was done for WC Pay for it's integration). 

I would suggest this is done as a follow-up though to avoid this PR getting stale.

Reply:
This will be resolved by #1516. Not marking this as resolved for visibility.